include/pybind11/async.h

```diff
@@ -0,0 +1,142 @@
+#pragma once
+
+#include <memory>
+#include <future>
+#include <map>
+#include <algorithm>
+#include <sstream>
+
+#include <chrono>
+#include <iostream>
+
+#include "pybind11/pybind11.h"
+#include "pybind11/stl.h"
+#include "pybind11/embed.h"
+#include "pybind11/operators.h"
+
+
+namespace pybind11 {
+
+template<typename ResultType>
+class Awaitable : public std::enable_shared_from_this<Awaitable<ResultType>>{
+    public:
+        Awaitable() {
+            this->future = std::future<ResultType>();
+        };
+
+        Awaitable(std::future<ResultType>& _future) {
+            this->future = std::move(_future);
+        };
+
+        std::shared_ptr<Awaitable<ResultType>> __iter__() {
```

Comment:
This is bad practice in C++, no? Aren't double underscores reserved for the language/compiler or so?

Reply:
right, i will change it and just leave the py interface with double underscores!